src/datasource.js

```diff
@@ -176,7 +176,13 @@ export class HawkularDatasource {
       return flatTags.map(tag => {
         return {text: tag, value: tag};
       });
-    });
+    }));
+  }
+
+  runWithResolvedVariables(target, func) {
+    const resolved = this.variables.resolve(target, {});
+    return this.q.all(resolved.map(p => func(p)))
+      .then(result => [].concat.apply([], result));
```

Comment:
What is the reason of returning result and empty array? I think the then can be completely omitted (since the `then` part will return copy of `result` array), if not you might consider using [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_operator)
```javascript
return this.q.all(resolved.map(p => func(p)))
  then(result => [...result]);
```